### ISSUE TITLE
Keyboard and emoji selection show up stacked together

### ISSUE THREAD

**Issue by major-mayer** [signalapp/Signal-Android#11618]
- [ x] I have searched open and closed issues for duplicates
- [ x] I am submitting a bug report for existing functionality that does not work as intended
- [ x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [ x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I have somehow managed to open the keyboard and the emojis altogether.
I can't really describe how i did this, but I got a screenshot and attached the debug logs, I hope this will be enough to find the reason.
This is the first time that i experienced such a behavior. Must be a real edge case that is happening rarely.

### Steps to reproduce
- Open Keyboard
- Do some weird stuff like opening the emojis and go back more or less at once

**Actual result:** Keyboard and emoji selector show up stacked 
**Expected result:** There should always only be one of those two: Keyboard or emojis

### Screenshots
![Screenshot_20210914-215748~2](https://user-images.githubusercontent.com/17907799/133339658-6750b54b-7c31-406a-a398-58e25ed1ccf5.png)

### Device info

**Device:** POCO F3
**Android version:** 11
**Signal version:** 5.23.6

### Link to debug log
https://debuglogs.org/3dc56e79cb88988890fbdeb3ab0f5541e3f5907fbfd7221b6c01f30939c827a6

**Comment by lennartkloock** [comment]
I can confirm this. But it only happened to me once or twice and I can't reproduce it.

**Comment by greyson-signal** [comment]
Yeah, I've seen this happen to me once randomly but I can't manage to reproduce it :/ If anyone finds a way to force it to happen, please details some steps :)

**Comment by major-mayer** [comment]
I found a way how to reproduce this :)
- Write something in the chat 
- switch from keyboard to emojis.
- Instead of selecting an emoji, select a word in the textbox
-> The keyboard appears under the emojis.

But I am not exactly sure if this is the same bug that I had before, because in the screenshot that i posted, i have not selected something in the Textbox.
Still it could maybe help finding the error

**Comment by major-mayer** [comment]
Yes still relevant and occurs more often than I would like...

**Comment by major-mayer** [comment]
The other issue from above fixed this problem hopefully. Otherwise I will reopen it again

**Comment by major-mayer** [comment]
A wait the fix was reverted...

**Comment by major-mayer** [comment]
Still a problem

**Comment by major-mayer** [comment]
Still a problem, and i found a way on how to reproduce this:

1. Attach a image to a message
2. Write some caption on that image
3. Press the emoji icon (to add an emoji)
4. Move the blinking text cursor (?) on another position
5. Keyboard appears under the still open emoji selection
![Screenshot_20220727-215634_Signal~2](https://user-images.githubusercontent.com/17907799/181361089-275e45ef-d0c6-44db-a766-dc3f56bd6586.png)

**Comment by major-mayer** [comment]
Could you have a look at this long-standing and very annoying bug @greyson-signal ?

**Comment by cody-signal** [comment]
Thank you for the repro steps, very helpful!

**Comment by major-mayer** [comment]
Thank you @greyson-signal for the fix :heart:

**Comment by major-mayer** [comment]
![Screenshot_20221227-000300_Signal~2](https://user-images.githubusercontent.com/17907799/209587836-2bd99d39-951e-4daa-96c6-51557a01c9de.png)
Unfortunately the problem seems to be back on 6.6.2 @greyson-signal
Haven't found a way to reproduce this yet, but it appeared multiple times. 

https://debuglogs.org/android/6.6.2/0fb132d49d2fca86ef339e3c145e35afaea20476b00520ebed53a45888deb5d2

**Comment by greyson-signal** [comment]
I've re-opened, but unfortunately I can't repro either :( None of the previous ways this would happen (notably, selecting text in the compose box while the emoji keyboard is open) repro this. Let me know if you figure out a way!

**Comment by major-mayer** [comment]
Thanks, I will keep an eye open on this.
